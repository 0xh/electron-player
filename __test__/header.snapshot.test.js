/**
 * @author arman
 * @since 8/25/17
 *
 */

import React from 'react';
import {shallow} from 'enzyme';
import Header from '../app/components/dumb/header.component';

test('Header component renders menu, title & drawer components inside it correctly', () => {
  const wrapper = shallow(
    <Header
      tableView={false}
      drawerOpen={false}
      toggleTableView={jest.fn()}
      handleDrawerOpen={jest.fn()}
      handleDialogOpen={jest.fn()}
      handleCatViewChange={jest.fn()} />
  );

  expect(wrapper).toMatchSnapshot();
});
