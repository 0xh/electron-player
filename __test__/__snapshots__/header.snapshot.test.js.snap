// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders menu, title & drawer components inside it correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Paper
    circle={false}
    className="header__wrapper"
    rounded={true}
    transitionEnabled={true}
    zDepth={5}
>
    <div
        className="player__header"
    >
        <Menu
            drawerOpen={false}
            handleCatViewChange={[Function]}
            handleDrawerOpen={[Function]}
            tableView={false}
            toggleTableView={[Function]}
        />
        <Title
            text="Electron Player"
        />
        <Drawer
            disableSwipeToOpen={false}
            docked={false}
            onRequestChange={[Function]}
            open={false}
            openSecondary={true}
            swipeAreaWidth={30}
            width={200}
            zDepth={2}
        >
            <FlatButton
                className="drawer__hide"
                disabled={false}
                fullWidth={false}
                icon={<NavigationCancel />}
                label="Cancel"
                labelPosition="before"
                labelStyle={Object {}}
                onClick={[Function]}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
                style={
                    Object {
                        "textAlign": "left",
                        "width": "100%",
                      }
                }
            />
            <FlatButton
                className="open__library__dialog"
                disabled={false}
                fullWidth={false}
                icon={<AvLibraryMusic />}
                label="Music Library"
                labelPosition="before"
                labelStyle={Object {}}
                onClick={[Function]}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
                style={
                    Object {
                        "textAlign": "left",
                        "width": "100%",
                      }
                }
            />
        </Drawer>
    </div>
</Paper>,
  "nodes": Array [
    <Paper
      circle={false}
      className="header__wrapper"
      rounded={true}
      transitionEnabled={true}
      zDepth={5}
>
      <div
            className="player__header"
      >
            <Menu
                  drawerOpen={false}
                  handleCatViewChange={[Function]}
                  handleDrawerOpen={[Function]}
                  tableView={false}
                  toggleTableView={[Function]}
            />
            <Title
                  text="Electron Player"
            />
            <Drawer
                  disableSwipeToOpen={false}
                  docked={false}
                  onRequestChange={[Function]}
                  open={false}
                  openSecondary={true}
                  swipeAreaWidth={30}
                  width={200}
                  zDepth={2}
            >
                  <FlatButton
                        className="drawer__hide"
                        disabled={false}
                        fullWidth={false}
                        icon={<NavigationCancel />}
                        label="Cancel"
                        labelPosition="before"
                        labelStyle={Object {}}
                        onClick={[Function]}
                        onKeyboardFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onTouchStart={[Function]}
                        primary={false}
                        secondary={false}
                        style={
                              Object {
                                    "textAlign": "left",
                                    "width": "100%",
                                  }
                        }
                  />
                  <FlatButton
                        className="open__library__dialog"
                        disabled={false}
                        fullWidth={false}
                        icon={<AvLibraryMusic />}
                        label="Music Library"
                        labelPosition="before"
                        labelStyle={Object {}}
                        onClick={[Function]}
                        onKeyboardFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onTouchStart={[Function]}
                        primary={false}
                        secondary={false}
                        style={
                              Object {
                                    "textAlign": "left",
                                    "width": "100%",
                                  }
                        }
                  />
            </Drawer>
      </div>
</Paper>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Header
        drawerOpen={false}
        handleCatViewChange={[Function]}
        handleDialogOpen={[Function]}
        handleDrawerOpen={[Function]}
        tableView={false}
        toggleTableView={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "drawerOpen": false,
          "handleCatViewChange": [Function],
          "handleDialogOpen": [Function],
          "handleDrawerOpen": [Function],
          "tableView": false,
          "toggleTableView": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Paper
          circle={false}
          className="header__wrapper"
          rounded={true}
          transitionEnabled={true}
          zDepth={5}
>
          <div
                    className="player__header"
          >
                    <Menu
                              drawerOpen={false}
                              handleCatViewChange={[Function]}
                              handleDrawerOpen={[Function]}
                              tableView={false}
                              toggleTableView={[Function]}
                    />
                    <Title
                              text="Electron Player"
                    />
                    <Drawer
                              disableSwipeToOpen={false}
                              docked={false}
                              onRequestChange={[Function]}
                              open={false}
                              openSecondary={true}
                              swipeAreaWidth={30}
                              width={200}
                              zDepth={2}
                    >
                              <FlatButton
                                        className="drawer__hide"
                                        disabled={false}
                                        fullWidth={false}
                                        icon={<NavigationCancel />}
                                        label="Cancel"
                                        labelPosition="before"
                                        labelStyle={Object {}}
                                        onClick={[Function]}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onTouchStart={[Function]}
                                        primary={false}
                                        secondary={false}
                                        style={
                                                  Object {
                                                            "textAlign": "left",
                                                            "width": "100%",
                                                          }
                                        }
                              />
                              <FlatButton
                                        className="open__library__dialog"
                                        disabled={false}
                                        fullWidth={false}
                                        icon={<AvLibraryMusic />}
                                        label="Music Library"
                                        labelPosition="before"
                                        labelStyle={Object {}}
                                        onClick={[Function]}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onTouchStart={[Function]}
                                        primary={false}
                                        secondary={false}
                                        style={
                                                  Object {
                                                            "textAlign": "left",
                                                            "width": "100%",
                                                          }
                                        }
                              />
                    </Drawer>
          </div>
</Paper>,
        "_debugID": 2,
        "_renderedOutput": <Paper
          circle={false}
          className="header__wrapper"
          rounded={true}
          transitionEnabled={true}
          zDepth={5}
>
          <div
                    className="player__header"
          >
                    <Menu
                              drawerOpen={false}
                              handleCatViewChange={[Function]}
                              handleDrawerOpen={[Function]}
                              tableView={false}
                              toggleTableView={[Function]}
                    />
                    <Title
                              text="Electron Player"
                    />
                    <Drawer
                              disableSwipeToOpen={false}
                              docked={false}
                              onRequestChange={[Function]}
                              open={false}
                              openSecondary={true}
                              swipeAreaWidth={30}
                              width={200}
                              zDepth={2}
                    >
                              <FlatButton
                                        className="drawer__hide"
                                        disabled={false}
                                        fullWidth={false}
                                        icon={<NavigationCancel />}
                                        label="Cancel"
                                        labelPosition="before"
                                        labelStyle={Object {}}
                                        onClick={[Function]}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onTouchStart={[Function]}
                                        primary={false}
                                        secondary={false}
                                        style={
                                                  Object {
                                                            "textAlign": "left",
                                                            "width": "100%",
                                                          }
                                        }
                              />
                              <FlatButton
                                        className="open__library__dialog"
                                        disabled={false}
                                        fullWidth={false}
                                        icon={<AvLibraryMusic />}
                                        label="Music Library"
                                        labelPosition="before"
                                        labelStyle={Object {}}
                                        onClick={[Function]}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onTouchStart={[Function]}
                                        primary={false}
                                        secondary={false}
                                        style={
                                                  Object {
                                                            "textAlign": "left",
                                                            "width": "100%",
                                                          }
                                        }
                              />
                    </Drawer>
          </div>
</Paper>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    drawerOpen={false}
    handleCatViewChange={[Function]}
    handleDialogOpen={[Function]}
    handleDrawerOpen={[Function]}
    tableView={false}
    toggleTableView={[Function]}
/>,
}
`;
